pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -487,3 +487,8 @@ func Convert_networking_IngressSpec_To_v1beta1_IngressSpec(in *networking.Ingres
 	}
 	return nil
 }
+
+func Convert_networking_HTTPIngressPath_To_v1beta1_HTTPIngressPath(in *networking.HTTPIngressPath, out *extensionsv1beta1.HTTPIngressPath, s conversion.Scope) error {
+	// this drops the new PathType field as there is no equivalent in v1beta1.
```

Comment:
This is not OK - must round-trip.

Reply:
Not sure what I was thinking here, thanks for catching this!